src/main/groovy/io/jenkins/infra/repository_permissions_updater/ArtifactoryAPI.groovy

```diff
@@ -117,12 +117,14 @@ abstract class ArtifactoryAPI {
      * Converts the provided base name (expected to be a GitHub repository name of the form 'org/name') to a user name
      * for a non-existing token user.
      *
+     * Artifactory has a limit of 58 characters for username
+     *
      * @link https://www.jfrog.com/confluence/display/JFROG/Access+Tokens#AccessTokens-SupportAuthenticationforNon-ExistingUsers
      * @param baseName
      * @return
      */
     static @NonNull String toTokenUsername(String baseName) {
-        return 'CD-for-' + baseName.replaceAll('[ /]', '__')
+        return sha256('CD-' + baseName).substring(0, 56)
```

Comment:
Open to suggestion 😅 But since description provides 1024 characters why bother with a limit of 58.